### DISCUSSION TITLE
Resize window with i3-msg

### DISCUSSION THREAD

**Discussion by akblakney** [i3/i3#5596] [Q&A]
I am trying to write a script that sets up a layout on a workstation. As a basic example, I'd like it to open two terminal windows, and then resize the second one to be slightly larger. Launching the two terminals is easy enough:

`i3-msg "exec --no-startup-id kitty; exec --no-startup-id kitty; "`

But now, if I try to resize the last one as follows, I get an error:

`i3-msg "exec --no-startup-id kitty; exec --no-startup-id kitty; resize grow width 50 px" `

The error that I get is this basically: `Failed to find appropriate tiling containers for resize operation.` So basically, it doesn't know which one to resize I assume. If this were done in the i3 config file, without i3-msg, this wouldn't be an issue, but I am trying to get it working with i3-msg. 

I have also tried doing this with `for_window ...`, but haven't been able to get it work. It seems this shouldn't even be necessary though. Is there a way for i3-msg to get the current window to resize like it would normally in the config file?

**Comment by laowalter** [comment]
this works for me:

i3-msg resize set width 1200 height 800 > /dev/null

**Comment by chupre** [comment]
came across with the same issue. First, make sure that the window is in the place that allows for the desired resize operation. Second, you should exec sleep before calling i3-msg, that worked for me. Here is my script for setting up a music workspace on startup for example (also added current workspace thing in case you do something when sleep is called):
```
#!/bin/bash
sleep 1
CURRENT_WORKSPACE=$(i3-msg -t get_workspaces \
  | jq '.[] | select(.focused==true).name' \
  | cut -d"\"" -f2)

i3-msg '[class="Yamusic"] focus; move left; move left; resize grow width 20 px or 20 ppt; workspace '"$CURRENT_WORKSPACE"
```
